api/valor_api/backend/core/llm_instructions_analysis.py

```diff
@@ -0,0 +1,573 @@
+def _generate_claims_instruction(text: str) -> str:
+    """
+    Instruction template was adapted from DeepEval's codebase https://github.com/confident-ai/deepeval/blob/main/deepeval/metrics/faithfulness/template.py.
+
+    Modifications to the instruction include improvements to the spelling, grammar, formatting and examples.
+
+    Parameters
+    ----------
+    text: str
+        The text to extract claims from.
+
+    Returns
+    -------
+    str
+        The instruction for the llm.
+    """
+    return f"""Based on the text, generate a comprehensive list of FACTUAL CLAIMS that can be inferred from the text.
+
+IMPORTANT: Return in JSON format with the "claims" key mapping to a list of strings. No words or explanation is needed.
+Only include claims that are factual. The claims you extract should include the full context it was presented in, NOT cherry picked facts.
+You should NOT include any prior knowledge. Take the text at face value when extracting claims.
+
+===== EXAMPLE ======
+Example Text: "Einstein won the noble prize in 1968 for his discovery of the photoelectric effect."
+
+Example JSON:
+{{
+    "claims": [
+        "Einstein won the noble prize for his discovery of the photoelectric effect.",
+        "Einstein won the noble prize in 1968."
+    ]
+}}
+===== END OF EXAMPLE ======
+
+Text:
+{text}
+
+JSON:
+"""
+
+
+def _generate_opinions_instruction(text: str) -> str:
+    """
+    Instruction template was adapted from DeepEval's codebase https://github.com/confident-ai/deepeval/blob/main/deepeval/metrics/bias/template.py.
+
+    Modifications to the instruction include improvements to the spelling, grammar, formatting and examples.
+
+    Parameters
+    ----------
+    text: str
+        The text to extract opinions from.
+
+    Returns
+    -------
+    str
+        The instruction for the llm.
+    """
+    return f"""Based on the text, generate a list of OPINIONS presented in the text. Claims and undisputed truths are NOT opinions.
+
+IMPORTANT: Return in JSON format with the "opinions" key mapping to a list of strings. No words or explanation is needed.
+Cited opinions should NOT be included as they are not opinions of the author of the text.
+Incorrect facts do NOT count as opinions.
+
+===== EXAMPLE ======
+Example Text: "Hitler hated jews, but I think the hate is unwarranted. Fox News thinks Donald Trump is a better President than Joe Biden. Earth is the smallest planet in our solar system."
```

Comment:
nit: some of these examples are a bit edgier than what I would include in a public repo.

Reply:
Agreed. The edgier parts are only there because I copied them over from DeepEval. I did another pass through and removed the edgier examples.